### ISSUE TITLE
Short-fn and fn differing behavior

### ISSUE THREAD

**Issue by andrewchambers** [janet-lang/janet#463]
```

# compiles
(fn [v] (let [x v] ~{:x ,x}))

# doesn't compile
|(let [x $] ~{:x ,x})

```

Found by @kamisori

**Comment by pepe** [comment]
Throws this error:

```
error: (macro) expected even number of arguments on line 1, column 1 while compiling repl
  in struct
  in qq [boot.janet] on line 1814, column 17
  in expandqq [boot.janet] (tailcall) on line 1816, column 21
  in macex1 [boot.janet] (tailcall) on line 1845, column 16
  in map [boot.janet] on line 868, column 36
  in expandall [boot.janet] (tailcall) on line 1791, column 15
  in macex1 [boot.janet] on line 1845, column 16
  in macex [boot.janet] on line 1918, column 18
  in short-fn [boot.janet] on line 1983, column 17
```

**Comment by bakpakin** [comment]
Thanks for finding this, this was a long standing bug in macex1, which was probably only excersized frequently by the short-fn macro.
